infrastructure/src/main/java/org/corfudb/infrastructure/BatchProcessor.java

```diff
@@ -114,127 +102,152 @@ private void recordRunnable(Runnable fsyncRunnable, Optional<Timer> fsyncTimer)
+                                streamLog.prefixTrim(addr);
                                 break;
                             case WRITE:
-                                WriteRequest write = (WriteRequest) currOp.getMsg().getPayload();
-                                Runnable append =
-                                        () -> streamLog.append(write.getGlobalAddress(), (LogData) write.getData());
+                                LogData logData = getLogData(payload.getWriteLogRequest().getLogData());
+                                Runnable append = () -> streamLog.append(logData.getGlobalAddress(), logData);
                                 recordRunnable(append, writeRecordsTimer);
                                 break;
                             case RANGE_WRITE:
-                                RangeWriteMsg writeRange = (RangeWriteMsg) currOp.getMsg().getPayload();
-                                Runnable appendMultiple = () -> streamLog.append(writeRange.getEntries());
+                                List<LogData> range = payload.getRangeWriteLogRequest().getLogDataList()
+                                        .stream().map(CorfuProtocolLogData::getLogData).collect(Collectors.toList());
+                                Runnable appendMultiple = () -> streamLog.append(range);
                                 recordRunnable(appendMultiple, writeRecordsTimer);
                                 break;
                             case RESET:
                                 streamLog.reset();
                                 break;
                             case TAILS_QUERY:
-                                TailsRequest tailsRequest = (TailsRequest)currOp.getMsg().getPayload();
-                                TailsResponse tails;
+                                final TailsResponse tails;
 
-                                switch (tailsRequest.getReqType()) {
-                                    case TailsRequest.LOG_TAIL:
+                                switch (payload.getTailRequest().getReqType()) {
+                                    case LOG_TAIL:
                                         tails = new TailsResponse(streamLog.getLogTail());
                                         break;
-
-                                    case TailsRequest.STREAMS_TAILS:
-                                        tails = streamLog.getTails(tailsRequest.getStreams());
+                                    case STREAMS_TAILS:
+                                        tails = streamLog.getTails(currentOp.getRequest()
+                                                .getPayload()
+                                                .getTailRequest()
+                                                .getStreamList()
+                                                .stream()
+                                                .map(CorfuProtocolCommon::getUUID)
+                                                .collect(Collectors.toList()));
```

Comment:
Good catch. It seems like this should be `ALL_STREAMS_TAIL`. This should be a bug, not sure how the tests are not failing. On reconfiguration, the sequencer won't be configured unless `LogUnitClient::getAllTails` is called, and that won't construct a STREAMS_TAILS type.

Reply:
I see. Its not failing because its hitting the default cause which returns all stream tails. I would recommend throwing an exception in the default case and explicitly specifying `ALL_STREAMS_TAIL`

Reply:
I think that this means we can remove the `STREAMS_TAILS` case entirely, since this is the only place where it's referenced, and the client never sends this type of message.

Reply:
Removed STREAMS_TAILS and related codes, thanks!